### ISSUE TITLE
`terraform_required_providers`: allow missing `version` for built-in providers

### ISSUE THREAD

**Issue by drewmullen** [terraform-linters/tflint#1280]
`terraform test` does not support a `version` parameter [yet](https://github.com/hashicorp/terraform/issues/30164). currently tflint is warning, failing CI (see below). I believe the `tflint-ignore` is not taking effect because there is actually no line to ignore. It would be nice if there was a workaround for situations like this. If im missing something please lmk!

```hcl
terraform {
  required_version = ">= 1.1.0"

  required_providers {
    test = {
      # tflint-ignore: terraform_required_providers
      source = "terraform.io/builtin/test"
    }
}
```

```
Warning: Missing version constraint for provider "test" in "required_providers" (terraform_required_providers)

  on tests.tf line 16:
  16: resource "test_assertions" "ipam_vpc_explicit_netmask" {

Reference: https://github.com/terraform-linters/tflint/blob/v0.33.1/docs/rules/terraform_required_providers.md
```

## Version

```console
$ tflint -v
TFLint version 0.33.1
WARNING: The plugin `aws` is not explicitly enabled. The bundled plugin will be enabled instead, but it is deprecated and will be removed in a future version. Please see https://github.com/terraform-linters/tflint/pull/1160 for details.
+ ruleset.aws (0.8.0-bundled)

$ tf -v  
Terraform v1.1.0
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.69.0
```

**Comment by wata727** [comment]
Annotation works for the reported line. In this case, you should add the `tflint-ignore` comment on `test.tf` line 15 or 16.
https://github.com/terraform-linters/tflint/blob/v0.33.1/docs/user-guide/annotations.md

**Comment by bendrucker** [comment]
Built in providers may not have a version distinct from the Terraform CLI version. Ignoring the builtin namespace is reasonable.

**Comment by drewmullen** [comment]
@wata727 is correct that ignoring the resource (rather than the provider block) does properly prevent the error. thank you!

@bendrucker is this a potential FR or should we close this issue? Im happy to reword the issue if you see a FR coming out of this

**Comment by bendrucker** [comment]
Leave it open, it's an easy PR for anyone and I'll get to it otherwise.
